### PR TITLE
remove uses of LINPACK since it has been defunct since R 3.1. Closes …

### PR DESCRIPTION
**Pull request by njtierney**
…#386

### CODE REVIEW

**Comment by goldingn** [comment]
Looks good! I'm just wondering if we can/should do more than R to let users know they are being ignored when specifying that. E.g. at least adding a details section to the function helpfile to confirm that, as in R, the LINPACK argument is defunct and silently ignored and the argument is only included for compatibility with the base function. What do you think?

---

Tangentially, I noticed that the docs for `base::chol2inv()` say:
> LINPACK   	logical. Defunct and ignored **(with a warning for true value)**.

(boldface is mine)

But it doesn't actually seem to warn in my version of R, e.g. if you do this:
```r
cma <- base::chol(ma  <- cbind(1, 1:3, c(1,3,7)))
base::chol2inv(cma, LINPACK = TRUE)
```

**Comment by njtierney** [comment]
I think it makes sense to include some more detail in the documentation, I'll add that now.

Mine gives a warning - could be a version 4.0.0 thing?

``` r
  cma <- base::chol(ma  <- cbind(1, 1:3, c(1,3,7)))
base::chol2inv(cma, LINPACK = TRUE)
#> Warning in base::chol2inv(cma, LINPACK = TRUE): the LINPACK argument has been
#> defunct since R 3.1.0
#>      [,1] [,2] [,3]
#> [1,]  2.5   -2  0.5
#> [2,] -2.0    3 -1.0
#> [3,]  0.5   -1  0.5
```

<sup>Created on 2021-04-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.5 (2021-03-31)
#>  os       macOS Big Sur 10.16         
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_AU.UTF-8                 
#>  ctype    en_AU.UTF-8                 
#>  tz       Australia/Perth             
#>  date     2021-04-27                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date       lib source            
#>  backports     1.2.1   2020-12-09 [1] standard (@1.2.1) 
#>  cli           2.4.0   2021-04-05 [1] CRAN (R 4.0.2)    
#>  crayon        1.4.1   2021-02-08 [1] CRAN (R 4.0.2)    
#>  digest        0.6.27  2020-10-24 [1] standard (@0.6.27)
#>  ellipsis      0.3.1   2020-05-15 [1] standard (@0.3.1) 
#>  evaluate      0.14    2019-05-28 [1] standard (@0.14)  
#>  fansi         0.4.2   2021-01-15 [1] CRAN (R 4.0.2)    
#>  fs            1.5.0   2020-07-31 [1] standard (@1.5.0) 
#>  glue          1.4.2   2020-08-27 [1] standard (@1.4.2) 
#>  highr         0.8     2019-03-20 [1] standard (@0.8)   
#>  htmltools     0.5.1.1 2021-01-22 [1] CRAN (R 4.0.2)    
#>  knitr         1.31    2021-01-27 [1] CRAN (R 4.0.2)    
#>  lifecycle     1.0.0   2021-02-15 [1] CRAN (R 4.0.2)    
#>  magrittr      2.0.1   2020-11-17 [1] standard (@2.0.1) 
#>  pillar        1.6.0   2021-04-13 [1] CRAN (R 4.0.2)    
#>  pkgconfig     2.0.3   2019-09-22 [1] standard (@2.0.3) 
#>  purrr         0.3.4   2020-04-17 [1] standard (@0.3.4) 
#>  reprex        2.0.0   2021-04-02 [1] CRAN (R 4.0.2)    
#>  rlang         0.4.10  2020-12-30 [1] CRAN (R 4.0.2)    
#>  rmarkdown     2.7     2021-02-19 [1] CRAN (R 4.0.2)    
#>  sessioninfo   1.1.1   2018-11-05 [1] standard (@1.1.1) 
#>  stringi       1.5.3   2020-09-09 [1] standard (@1.5.3) 
#>  stringr       1.4.0   2019-02-10 [1] standard (@1.4.0) 
#>  styler        1.4.1   2021-03-30 [1] CRAN (R 4.0.2)    
#>  tibble        3.1.1   2021-04-18 [1] CRAN (R 4.0.3)    
#>  utf8          1.2.1   2021-03-12 [1] CRAN (R 4.0.2)    
#>  vctrs         0.3.7   2021-03-29 [1] CRAN (R 4.0.3)    
#>  withr         2.4.2   2021-04-18 [1] CRAN (R 4.0.3)    
#>  xfun          0.22    2021-03-11 [1] CRAN (R 4.0.2)    
#>  yaml          2.2.1   2020-02-01 [1] standard (@2.2.1) 
#> 
#> [1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

</details>

**Comment by goldingn** [comment]
Aha. Interesting. In that case, to match R behaviour should we issue the same warning?

File: R/overloaded.R [hunk 1]
```diff
@@ -13,5 +13,10 @@
 #'   x,y,size,LINPACK,V,na.rm,dims,MARGIN,STATS,FUN,check.margin,\dots,r,k,upper.tri,transpose,l,X,INDEX,symmetric,only.values,EISPACK,x1,x2,compact,along,rev.along,new.names,force.array,make.names,use.anon.names,use.first.dimnames,hier.names,use.dnns,nrow,ncol
 #'    arguments as in original documentation
 #'
+#' @details
+#'   Note that, as in R, the LINPACK argument is defunct and silently ignored.
+#'   The argument is only included for compatibility with the base functions
+#'   that call it.
+#'
```
Comment (njtierney) [inline R/overloaded.R:20]: @goldingn I wasn't sure where to put this - does the `overloaded.R` file make sense, or should this go in the `functions.R` file, where `chol2inv` is exported? I just wasn't sure which function to provide the details to.

**Review by njtierney — COMMENTED** [review]
I don't think we need to though, it looks like R captures it?

``` r
library(greta)
#> 
#> Attaching package: 'greta'
#> The following objects are masked from 'package:stats':
#> 
#>     binomial, cov2cor, poisson
#> The following objects are masked from 'package:base':
#> 
#>     %*%, apply, backsolve, beta, chol2inv, colMeans, colSums, diag,
#>     eigen, forwardsolve, gamma, identity, rowMeans, rowSums, sweep,
#>     tapply
ma  <- cbind(1, 1:3, c(1,3,7))
ma_greta <- as_data(ma)
cma <- base::chol(ma)
cma_greta <- base::chol(ma_greta)
base::chol2inv(cma, LINPACK = TRUE)
#> Warning in base::chol2inv(cma, LINPACK = TRUE): the LINPACK argument has been
#> defunct since R 3.1.0
#>      [,1] [,2] [,3]
#> [1,]  2.5   -2  0.5
#> [2,] -2.0    3 -1.0
#> [3,]  0.5   -1  0.5
base::chol2inv(cma_greta, LINPACK = TRUE)
#> Warning in base::chol2inv(cma_greta, LINPACK = TRUE): the LINPACK argument has
#> been defunct since R 3.1.0
#>      [,1] [,2] [,3]
#> [1,]   NA   NA   NA
#> [2,]   NA   NA   NA
#> [3,]   NA   NA   NA
```

``` r
library(greta)
#> 
#> Attaching package: 'greta'
#> The following objects are masked from 'package:stats':
#> 
#>     binomial, cov2cor, poisson
#> The following objects are masked from 'package:base':
#> 
#>     %*%, apply, backsolve, beta, chol2inv, colMeans, colSums, diag,
#>     eigen, forwardsolve, gamma, identity, rowMeans, rowSums, sweep,
#>     tapply
ma  <- cbind(1, 1:3, c(1,3,7))
ma_greta <- as_data(ma)
cma <- base::chol(ma)
cma_greta <- base::chol(ma_greta)
base::chol2inv(cma, LINPACK = TRUE)
#> Warning in base::chol2inv(cma, LINPACK = TRUE): the LINPACK argument has been
#> defunct since R 3.1.0
#>      [,1] [,2] [,3]
#> [1,]  2.5   -2  0.5
#> [2,] -2.0    3 -1.0
#> [3,]  0.5   -1  0.5
base::chol2inv(cma_greta, LINPACK = TRUE)
#> Warning in base::chol2inv(cma_greta, LINPACK = TRUE): the LINPACK argument has
#> been defunct since R 3.1.0
#>      [,1] [,2] [,3]
#> [1,]   NA   NA   NA
#> [2,]   NA   NA   NA
#> [3,]   NA   NA   NA
```

<sup>Created on 2021-04-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.5 (2021-03-31)
#>  os       macOS Big Sur 10.16         
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_AU.UTF-8                 
#>  ctype    en_AU.UTF-8                 
#>  tz       Australia/Perth             
#>  date     2021-04-27                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version    date       lib source            
#>  backports     1.2.1      2020-12-09 [1] standard (@1.2.1) 
#>  base64enc     0.1-3      2015-07-28 [1] standard (@0.1-3) 
#>  cli           2.4.0      2021-04-05 [1] CRAN (R 4.0.2)    
#>  coda          0.19-4     2020-09-30 [1] CRAN (R 4.0.2)    
#>  codetools     0.2-18     2020-11-04 [1] CRAN (R 4.0.5)    
#>  crayon        1.4.1      2021-02-08 [1] CRAN (R 4.0.2)    
#>  digest        0.6.27     2020-10-24 [1] standard (@0.6.27)
#>  ellipsis      0.3.1      2020-05-15 [1] standard (@0.3.1) 
#>  evaluate      0.14       2019-05-28 [1] standard (@0.14)  
#>  fansi         0.4.2      2021-01-15 [1] CRAN (R 4.0.2)    
#>  fs            1.5.0      2020-07-31 [1] standard (@1.5.0) 
#>  future        1.21.0     2020-12-10 [1] CRAN (R 4.0.2)    
#>  globals       0.14.0     2020-11-22 [1] CRAN (R 4.0.2)    
#>  glue          1.4.2      2020-08-27 [1] standard (@1.4.2) 
#>  greta       * 0.3.1.9012 2021-04-22 [1] local             
#>  highr         0.8        2019-03-20 [1] standard (@0.8)   
#>  hms           1.0.0      2021-01-13 [1] CRAN (R 4.0.2)    
#>  htmltools     0.5.1.1    2021-01-22 [1] CRAN (R 4.0.2)    
#>  jsonlite      1.7.2      2020-12-09 [1] standard (@1.7.2) 
#>  knitr         1.31       2021-01-27 [1] CRAN (R 4.0.2)    
#>  lattice       0.20-41    2020-04-02 [1] CRAN (R 4.0.5)    
#>  lifecycle     1.0.0      2021-02-15 [1] CRAN (R 4.0.2)    
#>  listenv       0.8.0      2019-12-05 [1] CRAN (R 4.0.2)    
#>  magrittr      2.0.1      2020-11-17 [1] standard (@2.0.1) 
#>  Matrix        1.3-2      2021-01-06 [1] CRAN (R 4.0.5)    
#>  parallelly    1.24.0     2021-03-14 [1] CRAN (R 4.0.2)    
#>  pillar        1.6.0      2021-04-13 [1] CRAN (R 4.0.2)    
#>  pkgconfig     2.0.3      2019-09-22 [1] standard (@2.0.3) 
#>  prettyunits   1.1.1      2020-01-24 [1] standard (@1.1.1) 
#>  progress      1.2.2      2019-05-16 [1] CRAN (R 4.0.2)    
#>  purrr         0.3.4      2020-04-17 [1] standard (@0.3.4) 
#>  R6            2.5.0      2020-10-28 [1] standard (@2.5.0) 
#>  Rcpp          1.0.6      2021-01-15 [1] CRAN (R 4.0.2)    
#>  reprex        2.0.0      2021-04-02 [1] CRAN (R 4.0.2)    
#>  reticulate    1.18       2020-10-25 [1] CRAN (R 4.0.2)    
#>  rlang         0.4.10     2020-12-30 [1] CRAN (R 4.0.2)    
#>  rmarkdown     2.7        2021-02-19 [1] CRAN (R 4.0.2)    
#>  sessioninfo   1.1.1      2018-11-05 [1] standard (@1.1.1) 
#>  stringi       1.5.3      2020-09-09 [1] standard (@1.5.3) 
#>  stringr       1.4.0      2019-02-10 [1] standard (@1.4.0) 
#>  styler        1.4.1      2021-03-30 [1] CRAN (R 4.0.2)    
#>  tensorflow    2.4.0      2021-03-23 [1] CRAN (R 4.0.3)    
#>  tfruns        1.5.0      2021-02-26 [1] CRAN (R 4.0.2)    
#>  tibble        3.1.1      2021-04-18 [1] CRAN (R 4.0.3)    
#>  utf8          1.2.1      2021-03-12 [1] CRAN (R 4.0.2)    
#>  vctrs         0.3.7      2021-03-29 [1] CRAN (R 4.0.3)    
#>  whisker       0.4        2019-08-28 [1] standard (@0.4)   
#>  withr         2.4.2      2021-04-18 [1] CRAN (R 4.0.3)    
#>  xfun          0.22       2021-03-11 [1] CRAN (R 4.0.2)    
#>  yaml          2.2.1      2020-02-01 [1] standard (@2.2.1) 
#> 
#> [1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

</details>

**Comment by goldingn** [comment]
Try doing that to the Greta array with `chol2inv()` not `base::chol2inv()`?

File: R/overloaded.R (hunk 1 above)
Comment (goldingn) [reply R/overloaded.R:20]: functions.R I think, since that's where the helpfile info for that function is

**Comment by njtierney** [comment]
Ah yes that will do it

``` r
library(greta)
#> 
#> Attaching package: 'greta'
#> The following objects are masked from 'package:stats':
#> 
#>     binomial, cov2cor, poisson
#> The following objects are masked from 'package:base':
#> 
#>     %*%, apply, backsolve, beta, chol2inv, colMeans, colSums, diag,
#>     eigen, forwardsolve, gamma, identity, rowMeans, rowSums, sweep,
#>     tapply
ma  <- cbind(1, 1:3, c(1,3,7))
ma_greta <- as_data(ma)
cma <- base::chol(ma)
cma_greta <- base::chol(ma_greta)
base::chol2inv(cma, LINPACK = TRUE)
#> Warning in base::chol2inv(cma, LINPACK = TRUE): the LINPACK argument has been
#> defunct since R 3.1.0
#>      [,1] [,2] [,3]
#> [1,]  2.5   -2  0.5
#> [2,] -2.0    3 -1.0
#> [3,]  0.5   -1  0.5
chol2inv(cma_greta, LINPACK = TRUE)
#> greta array (operation)
#> 
#>      [,1] [,2] [,3]
#> [1,]  ?    ?    ?  
#> [2,]  ?    ?    ?  
#> [3,]  ?    ?    ?  
#> attr(,"class")
#> [1] matrix array
```

<sup>Created on 2021-04-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.5 (2021-03-31)
#>  os       macOS Big Sur 10.16         
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_AU.UTF-8                 
#>  ctype    en_AU.UTF-8                 
#>  tz       Australia/Perth             
#>  date     2021-04-27                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version    date       lib source            
#>  backports     1.2.1      2020-12-09 [1] standard (@1.2.1) 
#>  base64enc     0.1-3      2015-07-28 [1] standard (@0.1-3) 
#>  cli           2.4.0      2021-04-05 [1] CRAN (R 4.0.2)    
#>  coda          0.19-4     2020-09-30 [1] CRAN (R 4.0.2)    
#>  codetools     0.2-18     2020-11-04 [1] CRAN (R 4.0.5)    
#>  crayon        1.4.1      2021-02-08 [1] CRAN (R 4.0.2)    
#>  digest        0.6.27     2020-10-24 [1] standard (@0.6.27)
#>  ellipsis      0.3.1      2020-05-15 [1] standard (@0.3.1) 
#>  evaluate      0.14       2019-05-28 [1] standard (@0.14)  
#>  fansi         0.4.2      2021-01-15 [1] CRAN (R 4.0.2)    
#>  fs            1.5.0      2020-07-31 [1] standard (@1.5.0) 
#>  future        1.21.0     2020-12-10 [1] CRAN (R 4.0.2)    
#>  globals       0.14.0     2020-11-22 [1] CRAN (R 4.0.2)    
#>  glue          1.4.2      2020-08-27 [1] standard (@1.4.2) 
#>  greta       * 0.3.1.9012 2021-04-27 [1] local             
#>  highr         0.8        2019-03-20 [1] standard (@0.8)   
#>  hms           1.0.0      2021-01-13 [1] CRAN (R 4.0.2)    
#>  htmltools     0.5.1.1    2021-01-22 [1] CRAN (R 4.0.2)    
#>  jsonlite      1.7.2      2020-12-09 [1] standard (@1.7.2) 
#>  knitr         1.31       2021-01-27 [1] CRAN (R 4.0.2)    
#>  lattice       0.20-41    2020-04-02 [1] CRAN (R 4.0.5)    
#>  lifecycle     1.0.0      2021-02-15 [1] CRAN (R 4.0.2)    
#>  listenv       0.8.0      2019-12-05 [1] CRAN (R 4.0.2)    
#>  magrittr      2.0.1      2020-11-17 [1] standard (@2.0.1) 
#>  Matrix        1.3-2      2021-01-06 [1] CRAN (R 4.0.5)    
#>  parallelly    1.24.0     2021-03-14 [1] CRAN (R 4.0.2)    
#>  pillar        1.6.0      2021-04-13 [1] CRAN (R 4.0.2)    
#>  pkgconfig     2.0.3      2019-09-22 [1] standard (@2.0.3) 
#>  prettyunits   1.1.1      2020-01-24 [1] standard (@1.1.1) 
#>  progress      1.2.2      2019-05-16 [1] CRAN (R 4.0.2)    
#>  purrr         0.3.4      2020-04-17 [1] standard (@0.3.4) 
#>  R6            2.5.0      2020-10-28 [1] standard (@2.5.0) 
#>  Rcpp          1.0.6      2021-01-15 [1] CRAN (R 4.0.2)    
#>  reprex        2.0.0      2021-04-02 [1] CRAN (R 4.0.2)    
#>  reticulate    1.18       2020-10-25 [1] CRAN (R 4.0.2)    
#>  rlang         0.4.10     2020-12-30 [1] CRAN (R 4.0.2)    
#>  rmarkdown     2.7        2021-02-19 [1] CRAN (R 4.0.2)    
#>  sessioninfo   1.1.1      2018-11-05 [1] standard (@1.1.1) 
#>  stringi       1.5.3      2020-09-09 [1] standard (@1.5.3) 
#>  stringr       1.4.0      2019-02-10 [1] standard (@1.4.0) 
#>  styler        1.4.1      2021-03-30 [1] CRAN (R 4.0.2)    
#>  tensorflow    2.4.0      2021-03-23 [1] CRAN (R 4.0.3)    
#>  tfruns        1.5.0      2021-02-26 [1] CRAN (R 4.0.2)    
#>  tibble        3.1.1      2021-04-18 [1] CRAN (R 4.0.3)    
#>  utf8          1.2.1      2021-03-12 [1] CRAN (R 4.0.2)    
#>  vctrs         0.3.7      2021-03-29 [1] CRAN (R 4.0.3)    
#>  whisker       0.4        2019-08-28 [1] standard (@0.4)   
#>  withr         2.4.2      2021-04-18 [1] CRAN (R 4.0.3)    
#>  xfun          0.22       2021-03-11 [1] CRAN (R 4.0.2)    
#>  yaml          2.2.1      2020-02-01 [1] standard (@2.2.1) 
#> 
#> [1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

</details>

I'll add a warning message then - should we mimic the current R one?

**Comment by njtierney** [comment]
Perhaps something like this? ``` r
library(greta)
#> 
#> Attaching package: 'greta'
#> The following objects are masked from 'package:stats':
#> 
#>     binomial, cov2cor, poisson
#> The following objects are masked from 'package:base':
#> 
#>     %*%, apply, backsolve, beta, chol2inv, colMeans, colSums, diag,
#>     eigen, forwardsolve, gamma, identity, rowMeans, rowSums, sweep,
#>     tapply
ma  <- cbind(1, 1:3, c(1,3,7))
ma_greta <- as_data(ma)
cma <- base::chol(ma)
cma_greta <- base::chol(ma_greta)
base::chol2inv(cma, LINPACK = TRUE)
#> Warning in base::chol2inv(cma, LINPACK = TRUE): the LINPACK argument has been
#> defunct since R 3.1.0
#>      [,1] [,2] [,3]
#> [1,]  2.5   -2  0.5
#> [2,] -2.0    3 -1.0
#> [3,]  0.5   -1  0.5
chol2inv(cma_greta, LINPACK = TRUE)
#> Warning in chol2inv.greta_array(cma_greta, LINPACK = TRUE): The 'LINPACK'
#> argument is ignored for greta arrays, and has also been defunct since R 3.1.0
#> greta array (operation)
#> 
#>      [,1] [,2] [,3]
#> [1,]  ?    ?    ?  
#> [2,]  ?    ?    ?  
#> [3,]  ?    ?    ?  
#> attr(,"class")
#> [1] matrix array
```

<sup>Created on 2021-04-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.5 (2021-03-31)
#>  os       macOS Big Sur 10.16         
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_AU.UTF-8                 
#>  ctype    en_AU.UTF-8                 
#>  tz       Australia/Perth             
#>  date     2021-04-27                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version    date       lib source            
#>  backports     1.2.1      2020-12-09 [1] standard (@1.2.1) 
#>  base64enc     0.1-3      2015-07-28 [1] standard (@0.1-3) 
#>  cli           2.4.0      2021-04-05 [1] CRAN (R 4.0.2)    
#>  coda          0.19-4     2020-09-30 [1] CRAN (R 4.0.2)    
#>  codetools     0.2-18     2020-11-04 [1] CRAN (R 4.0.5)    
#>  crayon        1.4.1      2021-02-08 [1] CRAN (R 4.0.2)    
#>  digest        0.6.27     2020-10-24 [1] standard (@0.6.27)
#>  ellipsis      0.3.1      2020-05-15 [1] standard (@0.3.1) 
#>  evaluate      0.14       2019-05-28 [1] standard (@0.14)  
#>  fansi         0.4.2      2021-01-15 [1] CRAN (R 4.0.2)    
#>  fs            1.5.0      2020-07-31 [1] standard (@1.5.0) 
#>  future        1.21.0     2020-12-10 [1] CRAN (R 4.0.2)    
#>  globals       0.14.0     2020-11-22 [1] CRAN (R 4.0.2)    
#>  glue          1.4.2      2020-08-27 [1] standard (@1.4.2) 
#>  greta       * 0.3.1.9012 2021-04-27 [1] local             
#>  highr         0.8        2019-03-20 [1] standard (@0.8)   
#>  hms           1.0.0      2021-01-13 [1] CRAN (R 4.0.2)    
#>  htmltools     0.5.1.1    2021-01-22 [1] CRAN (R 4.0.2)    
#>  jsonlite      1.7.2      2020-12-09 [1] standard (@1.7.2) 
#>  knitr         1.31       2021-01-27 [1] CRAN (R 4.0.2)    
#>  lattice       0.20-41    2020-04-02 [1] CRAN (R 4.0.5)    
#>  lifecycle     1.0.0      2021-02-15 [1] CRAN (R 4.0.2)    
#>  listenv       0.8.0      2019-12-05 [1] CRAN (R 4.0.2)    
#>  magrittr      2.0.1      2020-11-17 [1] standard (@2.0.1) 
#>  Matrix        1.3-2      2021-01-06 [1] CRAN (R 4.0.5)    
#>  parallelly    1.24.0     2021-03-14 [1] CRAN (R 4.0.2)    
#>  pillar        1.6.0      2021-04-13 [1] CRAN (R 4.0.2)    
#>  pkgconfig     2.0.3      2019-09-22 [1] standard (@2.0.3) 
#>  prettyunits   1.1.1      2020-01-24 [1] standard (@1.1.1) 
#>  progress      1.2.2      2019-05-16 [1] CRAN (R 4.0.2)    
#>  purrr         0.3.4      2020-04-17 [1] standard (@0.3.4) 
#>  R6            2.5.0      2020-10-28 [1] standard (@2.5.0) 
#>  Rcpp          1.0.6      2021-01-15 [1] CRAN (R 4.0.2)    
#>  reprex        2.0.0      2021-04-02 [1] CRAN (R 4.0.2)    
#>  reticulate    1.18       2020-10-25 [1] CRAN (R 4.0.2)    
#>  rlang         0.4.10     2020-12-30 [1] CRAN (R 4.0.2)    
#>  rmarkdown     2.7        2021-02-19 [1] CRAN (R 4.0.2)    
#>  sessioninfo   1.1.1      2018-11-05 [1] standard (@1.1.1) 
#>  stringi       1.5.3      2020-09-09 [1] standard (@1.5.3) 
#>  stringr       1.4.0      2019-02-10 [1] standard (@1.4.0) 
#>  styler        1.4.1      2021-03-30 [1] CRAN (R 4.0.2)    
#>  tensorflow    2.4.0      2021-03-23 [1] CRAN (R 4.0.3)    
#>  tfruns        1.5.0      2021-02-26 [1] CRAN (R 4.0.2)    
#>  tibble        3.1.1      2021-04-18 [1] CRAN (R 4.0.3)    
#>  utf8          1.2.1      2021-03-12 [1] CRAN (R 4.0.2)    
#>  vctrs         0.3.7      2021-03-29 [1] CRAN (R 4.0.3)    
#>  whisker       0.4        2019-08-28 [1] standard (@0.4)   
#>  withr         2.4.2      2021-04-18 [1] CRAN (R 4.0.3)    
#>  xfun          0.22       2021-03-11 [1] CRAN (R 4.0.2)    
#>  yaml          2.2.1      2020-02-01 [1] standard (@2.2.1) 
#> 
#> [1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

</details>

**Comment by goldingn** [comment]
Yes, perfect!

**Comment by goldingn** [comment]
Great, that looks good to me. Assuming it passed all tests now?

File: R/overloaded.R (hunk 1 above)
Comment (njtierney) [reply R/overloaded.R:20]: Hmmm, when I search `?chol2inv` it gives an option to look for overloaded, so I think it has to go there?

```r
# nolint start
#' @rdname overloaded
#' @export
chol2inv <- function(x, size = NCOL(x), LINPACK = FALSE) {
  UseMethod("chol2inv", x)
}

```

**Comment by njtierney** [comment]
Yup, just an issue of where to put the `@details` comment, since it currently is in `overloaded.R` but adding the `@details` into the relevant part of `functions.R` doesn't seem to quite provide the details in the right spot.

**Comment by goldingn** [comment]
Your call on where to put it. The intention was that functions is the main documentation for these Greta versions, and overloaded is only there to stop R shouting about overloaded methods (we define a chol2inv generic so we can add a greta version).

I note that the call for chol2inv in functions doesn't list all the arguments though. Maybe the solution is to add all the arguments there and add details?

Alternatively, now we have the warning we maybe don't need the details at all?

**Pull request merged**